samcli/commands/build/build_context.py

```diff
@@ -155,7 +155,7 @@ def __exit__(self, *args):
     def get_resources_to_build(self):
         return self.resources_to_build
 
-    def run(self):
+    def run(self, sync=False):
```

Comment:
it is obvious but should we add typing here?

Reply:
Can we make the name more generic such as `show_message` instead of calling it `sync`?
In general, let's try to isolate different modules so that when going through BuildContext in the future, we don't need to see what sync means.

Another question, what's the reason for putting this parameter in run instead of `__init__`?